### PR TITLE
Print "Checking/Building ..." message even when --dry-run is passed

### PR DESCRIPTION
**Pull request by jyn514**
Print "Checking/Building ..." message even when --dry-run is passed
    
This makes it a lot easier to understand what commands will be run without
having to parse the `-vv` output, which isn't meant to be user facing.

I also want to change these messages at some point (https://github.com/rust-lang/rust/issues/102003) and this change will make it easier to paste a before/after comparison without having to actually build a stage 2 compiler.

### CODE REVIEW

File: src/bootstrap/config.rs [hunk 1]
```diff
@@ -84,7 +95,10 @@ pub struct Config {
     pub jobs: Option<u32>,
     pub cmd: Subcommand,
     pub incremental: bool,
-    pub dry_run: bool,
+    // #[cfg(not(test))]
+    // dry_run: DryRun,
+    // #[cfg(test)]
```
Comment (Mark-Simulacrum) [inline src/bootstrap/config.rs:100]: Stray comments?

**Review by Mark-Simulacrum — APPROVED** [review]
r=me with nit fixed

File: src/bootstrap/config.rs (hunk 1 above)
Comment (jyn514) [reply src/bootstrap/config.rs]: Yes, thanks :) I made this private so the compiler would suggest calling the new `dry_run()` function and I could use `cargo fix` to apply the suggestions.

**Pull request merged**